File: app/src/main/java/malayalamdictionary/samasya/adapter/ListItemAdapter.java
package malayalamdictionary.samasya.adapter;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v7.widget.CardView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Filter;
import android.widget.Filterable;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import malayalamdictionary.samasya.MainActivity;
import malayalamdictionary.samasya.R;

public class ListItemAdapter extends BaseAdapter implements Filterable {

    Context mContext;
    ArrayList<String> items;
    public static TextView textViewAutoComplete;
    private List<String> filteredItems;
    private ItemFilter mFilter = new ItemFilter();
    MainActivity mainActivity;
    CardView cardViewSuggestion;


    public ListItemAdapter(Context context, ArrayList<String> items, MainActivity mainActivity, CardView cardViewSuggestion){

        mContext=context;
        this.items=items;
        this.filteredItems = items;
        this.mainActivity=mainActivity;
        this.cardViewSuggestion=cardViewSuggestion;
    }

    @Override
    public int getCount() {
        return filteredItems.size();
    }

    @Override
    public Object getItem(int position) {
        return filteredItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {



        if (convertView==null){
            LayoutInflater layoutInflater= (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView=layoutInflater.inflate(R.layout.autocompletion_list,parent,false);
        }
        textViewAutoComplete= (TextView) convertView.findViewById(R.id.textView_autocomplete);

        Typeface type = Typeface.createFromAsset(mContext.getAssets(), "fonts/mal.ttf");
        if (MainActivity.englishToMayalayam){
            textViewAutoComplete.setTypeface(Typeface.DEFAULT);
        }
        else {
            textViewAutoComplete.setTypeface(type);
        }
//        textViewAutoComplete.setTypeface(type);
        String location = filteredItems.get(position);
        if (!location.isEmpty()) {

            textViewAutoComplete.setText(location);
        }
        else {

        }


            textViewAutoComplete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                mainActivity.fillData(filteredItems.get(position));
            }
        });
        return convertView;
    }

    private class ItemFilter extends Filter {
        @Override
        protected FilterResults performFiltering(CharSequence constraint) {
            String filterString = constraint.toString().toLowerCase();
            FilterResults results = new FilterResults();

            int count = items.size();
            final List<String> tempItems = new ArrayList<>(count);

            for (int i = 0; i < count; i++) {
                if (items.get(i).toLowerCase().contains(filterString)) {
                    tempItems.add(items.get(i));
                }
            }

            results.values = tempItems;
            results.count = tempItems.size();

            return results;
        }

        @Override
        protected void publishResults(CharSequence constraint, FilterResults results) {
            filteredItems = (ArrayList<String>) results.values;
            notifyDataSetChanged();
        }
    }

    public Filter getFilter() {
        return mFilter;
    }

}

File: app/src/main/java/malayalamdictionary/samasya/adapter/ListItemAdapter.kt
package malayalamdictionary.samasya.adapter

import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Filter
import android.widget.Filterable
import android.widget.TextView
import malayalamdictionary.samasya.MainActivity
import malayalamdictionary.samasya.R
import malayalamdictionary.samasya.helper.Common
import java.util.ArrayList

class ListItemAdapter(context: Context,items: ArrayList<String>, mainActivity: MainActivity): BaseAdapter(), Filterable{

    private val context=context
    private var items=items
    lateinit var textViewAutoComplete: TextView
    private var filteredItems: List<String> = items
    private var mainActivity: MainActivity =mainActivity
    private val mFilter = ItemFilter()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View? {

        var mConvertView=convertView
        if (mConvertView == null) {
            val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            mConvertView = layoutInflater.inflate(R.layout.autocompletion_list, parent, false)
        }
        textViewAutoComplete = mConvertView!!.findViewById(R.id.textView_autocomplete)

        val type = Typeface.createFromAsset(context.getAssets(), "fonts/mal.ttf")
        if (Common.englishToMayalayam) {
            textViewAutoComplete.typeface = Typeface.DEFAULT
        } else {
            textViewAutoComplete.typeface = type
        }
        val location = filteredItems[position]
        if (location.isNotEmpty()) {
            textViewAutoComplete.text = location
        }

        textViewAutoComplete.setOnClickListener { mainActivity.fillData(filteredItems[position]) }
        return mConvertView
    }

    override fun getItem(position: Int): Any {
        return filteredItems[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return filteredItems.size
    }

    private inner class ItemFilter : Filter() {
        override fun performFiltering(constraint: CharSequence): Filter.FilterResults {
            val filterString = constraint.toString().toLowerCase()
            val results = Filter.FilterResults()

            val count = items.size
            val tempItems = ArrayList<String>(count)

            for (i in 0 until count) {
                if (items[i].toLowerCase().contains(filterString)) {
                    tempItems.add(items[i])
                }
            }

            results.values = tempItems
            results.count = tempItems.size

            return results
        }

        override fun publishResults(constraint: CharSequence, results: Filter.FilterResults) {
            filteredItems = results.values as ArrayList<String>
            notifyDataSetChanged()
        }
    }
    override fun getFilter(): Filter {
        return mFilter
    }

}
